applications/ShapeOptimizationApplication/python_scripts/analyzer_internal.py

```diff
@@ -136,8 +141,10 @@ def __CreateResponseFunctions( specified_responses, model ):
                 if convdiff_response_factory is None:
                     raise RuntimeError("ShapeOpt: {} response function requires ConvectionDiffusionApplication.".format(response_type))
                 response_functions[response_id] = convdiff_response_factory.CreateResponseFunction(response_id, response_settings, model)
-            elif response_type in sho_response_functions:
-                response_functions[response_id] = sho_response_factory.CreateResponseFunction(response_id, response_settings, model)
+            elif response_type in cps_response_functions:
+                if potential_flow_response_factory is None:

```

Comment:
why not doing the import here?
This way you can avoid the try...except which we tried to remove since it could give issues sometimes
=> the general idea is to import only the apps you need

@KratosMultiphysics/optimization guess this is more a comment for you, seems to me it would be an easy refactor

Reply:
It would be easy, back then i followed the idea of having all imports at the top of the file. I am fine with any of the approaches.

Reply:
ah ok I understand, some time ago we had the issue of variable keys being assigned at runtime and not at compiletime
this should no longer be an issue, hence you could refactor

Reply:
I keep this in mind, but i would not do it in this PR.